packages/core/src/App/Containers/Routes/routes.jsx

```diff
@@ -34,21 +45,71 @@ class Routes extends React.Component {
         this.props.setAppRouterHistory(this.props.history);
     }
 
+    componentDidUpdate(pp, ps) {
+        if (pp.location.pathname !== this.props.location.pathname) {
+            this.isBot(this.props.location);
+        }
+        if (this.state.is_bot !== ps.is_bot) {
+            if (this.state.is_bot) {
+                this.removeBlockerEvents();
+                this.el_landscape_blocker.classList.add('landscape-blocker--keyboard-visible');
+            } else {
+                this.addBlockerEvents();
+                this.el_landscape_blocker.classList.remove('landscape-blocker--keyboard-visible');
+            }
+        }
+    }
+
     componentWillUnmount() {
+        this.removeBlockerEvents();
         if (typeof this.unlisten_to_change === 'function') {
             this.unlisten_to_change();
             this.unlisten_to_change = null;
             this.initial_route = null;
         }
     }
 
+    isBot = location => {
```

Comment:
Just a question, don't we usually have `const` or `function` for functions? These functions don't have that, just wanted to know why.

Reply:
I think, when we use function or constant inside a class we don't need const or function term to define